src/fastertransformer/models/t5/T5Decoding.cc

```diff
@@ -901,21 +919,23 @@ void T5Decoding<T>::forward(TensorMap*                 output_tensors,
                                   1,
                                   stream_);
         }
-        if (output_tensors->isExist("output_log_probs")) {
-            invokeTransposeAxis01(output_tensors->at("output_log_probs").getPtr<float>(),
-                                  output_log_probs_buf_,
-                                  max_seq_len,
-                                  batch_size * beam_width,
-                                  1,
-                                  stream_);
-        }
 
-        // Return the cumulative log probability if requested.
-        if (output_tensors->isExist("cum_log_probs")) {
-            Tensor cum_log_probs = output_tensors->at("cum_log_probs");
-            FT_CHECK_WITH_INFO(cum_log_probs.size() == batch_size * beam_width,
-                               "The shape of cum_log_probs does not match with batch_size x beam_width.");
-            cudaD2Dcpy(cum_log_probs.getPtr<float>(), cum_log_probs_, batch_size * beam_width);
+        // Return the cumulative log probability and log probability if requested.
+        if (!using_beam_hyps) {
```

Comment:
this seems to cause greedy search or sampling decoding not able to return output_log_probs or cum_log_probs since `using_beam_hyps` would be true for greedy search or sampling as well.